python/semantic_kernel/connectors/ai/ollama/services/ollama_text_embedding.py

```diff
@@ -2,51 +2,94 @@
     """
 
-    url: HttpUrl = "http://localhost:11434/api/embeddings"
-    session: aiohttp.ClientSession | None = None
+    def __init__(
+        self,
+        service_id: str | None = None,
+        ai_model_id: str | None = None,
+        host: str | None = None,
+        client: AsyncClient | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+    ) -> None:
+        """Initialize an OllamaChatCompletion service.
+
+        Args:
+            service_id (Optional[str]): Service ID tied to the execution settings. (Optional)
+            ai_model_id (Optional[str]): The model name. (Optional)
+            host (Optional[str]): URL of the Ollama server, defaults to None and
+                will use the default Ollama service address: http://127.0.0.1:11434. (Optional)
+            client (Optional[AsyncClient]): A custom Ollama client to use for the service. (Optional)
+            env_file_path (str | None): Use the environment settings file as a fallback to using env vars.
+            env_file_encoding (str | None): The encoding of the environment settings file, defaults to 'utf-8'.
+        """
+        try:
+            ollama_settings = OllamaSettings.create(
+                model=ai_model_id,
+                host=host,
+                env_file_path=env_file_path,
+                env_file_encoding=env_file_encoding,
+            )
+        except ValidationError as ex:
+            raise ServiceInitializationError("Failed to create Ollama settings.", ex) from ex
+
+        super().__init__(
+            service_id=service_id or ollama_settings.model,
+            ai_model_id=ollama_settings.model,
+            client=client or AsyncClient(host=ollama_settings.host),
+        )
 
     @override
-    async def generate_embeddings(self, texts: list[str], **kwargs: Any) -> ndarray:
+    async def generate_embeddings(
+        self,
+        texts: list[str],
+        settings: "PromptExecutionSettings | None" = None,
+        **kwargs: Any,
+    ) -> ndarray:
+        if not settings:
+            settings = OllamaEmbeddingPromptExecutionSettings()
+        else:
+            settings = self.get_prompt_execution_settings_from_settings(settings)
+
         result = []
         for text in texts:
-            async with (
-                AsyncSession(self.session) as session,
-                session.post(
-                    self.url,
-                    json={"model": self.ai_model_id, "prompt": text, "options": kwargs},
-                ) as response,
-            ):
-                response.raise_for_status()
-                response = await response.json()
-                result.append(response["embedding"])
+            response_object = await self.client.embeddings(
+                model=self.ai_model_id,
+                prompt=text,
+                **settings.prepare_settings_dict(),
+            )
+            result.append(response_object["embedding"])
```

Comment:
in current connectors, the internal embedding is also cast to a `array`, AFAIk there is no way to type hint this, but let's keep that consistent

Reply:
We introduced a `generate_raw_embeddings` method, you could use that to both return list[list[float]] and array